task/queue/queue.go

```diff
@@ -266,6 +266,10 @@ func (q *Queue) completeTask(ctx context.Context, tsk *task.Task) {
 	if err != nil {
 		logger.WithError(err).Error("Failure to update state during complete task")
 	}
+
+	if tsk.HasError() {
+		logger.WithError(tsk.Error.Error).Error("Error occurred while running task")
```

Comment:
Yes, it does. Perhaps not named the best, as it really should be something like `tsk.SerializableError.Error`, but its not worth the time right now to rename it across the code.

The reason why there are two layes here is because typically an `error` type object is not serializable (due to Golang limitations about what will and will not be serialized to JSON/BSON, such as private fields, etc.). So the outer "Error" is an object that can be serialized and knows how to pull out the relevant information from the contained `error` type to serialize that information.

I really wish we didn't need this extra layer, but I couldn't figure out a way around it without causing a number of more serious issues (specifically around Golang's interpretation of interfaces to nil concrete objects are not actually nil).